File: 6-zigzag-conversion/zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s

        i = 0
        direction = 1

        rows = [[] for _ in range(numRows)]

        for char in s:
            rows[i].append(char)
            if i == 0:
                d = 1
            elif i == numRows - 1:
                d = -1
            i += d

        ans = ''
        for i in range(numRows):
            ans += ''.join(rows[i])
        
        return ans
        

                

File: 6-zigzag-conversion/zigzag-conversion.java
class Solution {
    public String convert(String s, int numRows) {
        if (numRows == 1) {
            return s;
        }

        int i = 0;
        int d = 1;          

        StringBuilder[] rows = new StringBuilder[numRows];
        for (int j = 0; j < numRows; j++) {
            rows[j] = new StringBuilder();
        }

        for (char ch : s.toCharArray()) {
            rows[i].append(ch);

            if (i == 0) {
                d = 1;       
            } else if (i == numRows - 1) {
                d = -1;      
            }

            i += d;
        }

        StringBuilder ans = new StringBuilder();
        for (int j = 0; j < numRows; j++) {
            ans.append(rows[j]);
        }

        return ans.toString();
    }
}
